src/audio/google/google_ctc_audio_processing_mock.c

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Google LLC.
+//
+// Author: Eddy Hsu <eddyhsu@google.com>
+#include "google_ctc_audio_processing.h"
+#include "google_ctc_audio_processing_sof_message_reader.h"
+
+#include <rtos/alloc.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "ipc/topology.h"
+
+struct GoogleCtcAudioProcessingState {
+	int num_frames;
+	int partition_size;
+	int impulse_size;
+	int sample_rate;
+	int is_symmetric;
+};
+
+GoogleCtcAudioProcessingState *GoogleCtcAudioProcessingCreate(void)
+{
+	struct GoogleCtcAudioProcessingState *s =
+		rballoc(0, SOF_MEM_CAPS_RAM, sizeof(GoogleCtcAudioProcessingState));
+	if (!s)
+		return NULL;
+
+	s->num_frames = CONFIG_COMP_GOOGLE_CTC_AUDIO_PROCESSING_NUM_FRAMES;
+	s->partition_size = 64;
+	s->impulse_size = 256;
+	s->sample_rate = 48000;
+	s->is_symmetric = 0;
+	return s;
+}
+
+GoogleCtcAudioProcessingState *GoogleCtcAudioProcessingCreateWithConfig(
+	int num_frames, int partition_size, int impulse_size, int sample_rate,
+	int is_symmetric, const uint8_t *config, int config_size)
+{
+	struct GoogleCtcAudioProcessingState *s =
+		rballoc(0, SOF_MEM_CAPS_RAM, sizeof(GoogleCtcAudioProcessingState));
+	if (!s)
+		return NULL;
+
+	s->num_frames = num_frames;
+	s->partition_size = partition_size;
+	s->impulse_size = impulse_size;
+	s->sample_rate = sample_rate;
+	s->is_symmetric = is_symmetric;
+	return s;
+}
+
+void GoogleCtcAudioProcessingFree(GoogleCtcAudioProcessingState *state)
+{
+	rfree(state);
+}
+
+void GoogleCtcAudioProcessingProcess(GoogleCtcAudioProcessingState *state,
+				     const float *src, float *dest,
+				     int num_frames, int num_channels)
+{
```

Comment:
To repeat from elsewhere: note that the "state" parameter is ignored, so there's no point in putting any info in it.

Reply:
Removed.